### PR TITLE
Fix #12299: Change to POST request to avoid possibly very long URIs

### LINKED ISSUE #12299: Request-URI Too Long
### Steps to reproduce
1. Choose a database
2. Click the Search button and run a search through the database
3. Try and browse the results
### Expected behaviour

Should go to the found results
### Actual behaviour

Request-URI Too Long

The requested URL's length exceeds the capacity limit for this server.
### Server configuration

OSX

**Web server:**
Apache 2.4

**Database:**
MariaDB

**PHP version:**
5.6

**phpMyAdmin version:**
4.6.1
### Client configuration

**Browser:**
FF

**Operating system:**

### PR DESCRIPTION
**Pull request by devenbansod**
Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

Use custom attributes to embed the required information instead and gather it using JQuery while making the Post request. 

Fix #12299

### CODE REVIEW

File: libraries/DbSearch.php [hunk 1]
```diff
@@ -334,29 +334,21 @@ private function _getResultsRow($each_table, $newsearchsqls, $odd_row, $res_cnt)
...
-                    array(
-                        'db' => $GLOBALS['db'], 'table' => $each_table
-                    )
-                ) . '\''
-                . ');return false;" >'
+            $html_output .= '<td><a name="browse_search" '
+                . ' class="ajax browse_results" href="'
+                . $browse_result_path . '" '
+                . 'browse_sql="' . $newsearchsqls['select_columns'] . '" '
+                . 'table_name="' . $each_table . '" >'
```
Comment (nijel) [inline libraries/DbSearch.php]: Wouldn't be cleaner to use data-* attributes instead? (I'm not sure just asking)

File: libraries/DbSearch.php [hunk 2]
```diff
@@ -334,29 +334,21 @@ private function _getResultsRow($each_table, $newsearchsqls, $odd_row, $res_cnt)
...
-                    __('Delete the matches for the %s table?'),
-                    htmlspecialchars($each_table)
-                ))
-                . '\');return false;">'
+
+            $delete_result_path = $browse_result_path;
+            $html_output .= '<td><a name="delete_search" class="ajax delete_results"'
+                . ' href="' . $delete_result_path . '"'
+                . ' data-table-name="' . $each_table . '"'
+                . ' data-delete-sql="' . $newsearchsqls['delete'] . '" >'
```
Comment (nijel) [inline libraries/DbSearch.php]: Probably some htmlspecialchars are missing here as well. At least `$each_table` was previously escaped and now is not, but the SQL query and URL probably as well.

**Comment by nijel** [comment]
Thanks, merged!

**Comment by devenbansod** [comment]
Thanks for the review. :-)

**Comment by nijel** [comment]
Can you please merge this to master? There seem to be some non trivial conflicts and you now know the code better ;-).

**Comment by devenbansod** [comment]
It seems the I did not adjust the tests after the changes. I will do it in sometime.

**Comment by nijel** [comment]
I've fixed the tests (and generated markup) in 59b2c5a. The only remaining part is merging QA_4_6 to master.

**Pull request merged**